### PR TITLE
Search UI on mobile

### LINKED ISSUE #414: Search UI on mobile
Hasjob's filter toolbar is visible on all pages across the site, encouraging its use. On mobile however, it's hidden behind a magnifying glass icon at the top of the screen. Users may not notice this.

Proposal: have the keyword search field always visible at top of the screen. Change the placeholder text from "Keywords" to "Search". Tapping on the field should reveal the other filters, in a pane that slides down instead of sliding right.

### PR DESCRIPTION
**Pull request by vidya-ram**
search-box-mobile

Fixes #414

### CODE REVIEW

File: hasjob/static/js/app.js [hunk 1]
```diff
@@ -216,26 +194,45 @@ window.Hasjob.Filters = {
...
       data: this.getCurrentState(),
-      showSidebar: function() {
-        filters.ractive.set('sidebarOn', true);
+      show: function() {
+        filters.dropdownMenu.set('show', true);
+      },
+      hide: function() {
+        filters.dropdownMenu.set('show', false);
+      },
+      openDropDownMenu: function(event) {
```
Comment (shreyas-satish) [inline hasjob/static/js/app.js]: Is this and `closeDropDownMenu` really required? Since it's mostly just calling `show` (and `hide`).

Comment (vidya-ram) [reply hasjob/static/js/app.js]: Fixed

File: hasjob/static/js/app.js [hunk 2]
```diff
@@ -216,26 +194,37 @@ window.Hasjob.Filters = {
...
     var pageScrollTimerId;
 
-    filters.ractive = new Ractive({
+    filters.dropdownMenu = new Ractive({
       el: 'job-filters-ractive-template',
       template: '#filters-ractive',
       data: this.getCurrentState(),
-      showSidebar: function() {
-        filters.ractive.set('sidebarOn', true);
+      openDropDownMenu: function(event) {
```
Comment (shreyas-satish) [inline hasjob/static/js/app.js]: Just `open` and `close` are clear enough? Since the object is `dropdownMenu` anyway.

Comment (shreyas-satish) [reply hasjob/static/js/app.js]: Also, this is only applicable on mobile, right?

Comment (vidya-ram) [reply hasjob/static/js/app.js]: yes

Comment (shreyas-satish) [reply hasjob/static/js/app.js]: Maybe `revealOnMobile` is better?

File: hasjob/static/js/app.js [hunk 3]
```diff
@@ -216,26 +194,46 @@ window.Hasjob.Filters = {
       selectedCurrency: window.Hasjob.Config.selectedFilters.currency,
       pay: window.Hasjob.Config.selectedFilters.pay,
       equity: window.Hasjob.Config.selectedFilters.equity,
-      sidebarOn: false
+      isMobile: $(window).width() < 768
```
Comment (jace) [inline hasjob/static/js/app.js]: Is 768 a safe number to hardcode?

File: hasjob/static/js/app.js [hunk 4]
```diff
@@ -369,59 +365,14 @@ window.Hasjob.Filters = {
...
-        window.Hasjob.Body.ractive.on('swipeLeft', function() {
-          if (isSlidingMenu && filters.ButtonRactive.get('sidebarOn')) {
-            filters.ButtonRactive.hideSidebar();
-          }
-        });
+    //On pressing ESC, close the filters menu
+    $(document).keydown(function(event) {
+      if (event.keyCode === 27) {
+        event.preventDefault();
+        filters.dropdownMenu.closeOnMobile();
```
Comment (jace) [inline hasjob/static/js/app.js]: Have we covered tapping outside the filters box?

File: hasjob/static/js/app.js (hunk 3 above)
Comment (vidya-ram) [reply hasjob/static/js/app.js]: Our current layout, screen size >= 768px the filters menu is not a dropdown and is a horizontal bar below the header

File: hasjob/static/js/app.js (hunk 4 above)
Comment (vidya-ram) [reply hasjob/static/js/app.js]: Yes, https://github.com/hasgeek/hasjob/pull/429/files#diff-25f7a8564bafd30a472f661a106c6008R232

**Pull request merged**